### ISSUE TITLE
TypeScript documentation not updated for vue 3.3 typed slots

### ISSUE THREAD

**Issue by swantzter** [vuejs/docs#2371]
vuejs/core#7982 introduced typing of slots, and it was announced in the release post, however I can't find anything in the docs on how to use this new feature, not in the [slots](https://vuejs.org/guide/components/slots.html) documentation neither in the TypeScript ([composition](https://vuejs.org/guide/typescript/composition-api.html) / [options](https://vuejs.org/guide/typescript/options-api.html)) sections, the only information I found on how to use this new feature was in the PR itself.

It would be nice to get this added to the docs

**Comment by SimonSimCity** [comment]
The docs mention it here: https://vuejs.org/api/sfc-script-setup.html#defineslots 

But might be good to create a cross reference at the places you mentioned...

**Comment by swantzter** [comment]
Yeah, after searching a bit more I see the options API version is also available at https://vuejs.org/api/options-rendering.html#slots but neither this one nor the one you mentioned comes near the top when you use the search and search for "slots", or is linked to in those pages

**Comment by volarname** [comment]
what about example with no props ?

**Comment by volarname** [comment]
also looks like current code is mistake
documentation:
```
default(props: { msg: string }): any
```
should be:
```
default?: (props: { msg: string }) => any
```
or?

**Comment by jd-solanki** [comment]
> also looks like current code is mistake documentation:
> 
> ```
> default(props: { msg: string }): any
> ```
> 
> should be:
> 
> ```
> default?: (props: { msg: string }) => any
> ```
> 
> or?

Both are valid syntax. Please refer to [PR](https://github.com/vuejs/core/pull/7982) for more details.

Use `item?: (props: {}) => any` for no props.
